GitUI/UserControls/SortDiffListContextMenuItem.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using GitCommands;
+using GitUI.Properties;
+using ResourceManager;
+
+namespace GitUI.UserControls
+{
+    public class SortDiffListContextMenuItem : ToolStripMenuItem
+    {
+        private readonly TranslationString _sortByText = new TranslationString("&Sort by...");
+        private readonly TranslationString _filePathSortText = new TranslationString("File &Path");
+        private readonly TranslationString _fileExtensionSortText = new TranslationString("File &Extension");
+        private readonly TranslationString _fileStatusSortText = new TranslationString("File &Status");
+        private readonly IDiffListSortService _sortService;
+        private readonly ToolStripMenuItem _filePathSortItem;
+        private readonly ToolStripMenuItem _fileExtensionSortItem;
+        private readonly ToolStripMenuItem _fileStatusSortItem;
+
+        public SortDiffListContextMenuItem(IDiffListSortService sortService)
+        {
+            _sortService = sortService ?? throw new ArgumentNullException(nameof(sortService));
+            Image = Images.SortBy;
+            Text = _sortByText.Text;
+
+            _filePathSortItem = new ToolStripMenuItem()
+            {
+                Text = _filePathSortText.Text,
+                ShowShortcutKeys = true,
+                Image = null,
+                Tag = DiffListSortType.FilePath
+            };
+
+            _fileExtensionSortItem = new ToolStripMenuItem()
+            {
+                Text = _fileExtensionSortText.Text,
+                ShowShortcutKeys = true,
+                Image = null,
+                Tag = DiffListSortType.FileExtension
+            };
+
+            _fileStatusSortItem = new ToolStripMenuItem()
+            {
+                Text = _fileStatusSortText.Text,
+                ShowShortcutKeys = true,
+                Image = null,
+                Tag = DiffListSortType.FileStatus
+            };
+
+            foreach (var item in AllItems())
+            {
+                item.Click += Item_Click;
+                DropDownItems.Add(item);
+            }
+
+            RequerySortingMethod();
+        }
+
+        private ToolStripMenuItem[] AllItems()
```

Comment:
Please create a `private static readonly` variable and assign it in the ctor.
There is no need to allocate the array multiple times

Reply:
Will do.

Reply:
I assume you didn't mean static. Right?

Reply:
💭  I think it can be, the items are static and we can only show one instance of the menu at a time...
But what you have is good too. 

I'd change an array for a readonly list, as returning arrays is a bad practice (items can be replaced outside your control):
```suggestion
        private IReadOnlyList<ToolStripMenuItem> AllItems()
```